### PR TITLE
Feature/use phpmetrics

### PR DESCRIPTION
**Pull request by ERuban**

### CODE REVIEW

File: src/PhpMetricsChecker/Metric/CcnMethodMax.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+declare(strict_types=1);
+
+namespace EonX\EasyQuality\PhpMetricsChecker\Metric;
+
+final class CcnMethodMax extends AbstractMetric
```
Comment (roman-eonx) [inline src/PhpMetricsChecker/Metric/CcnMethodMax.php:6]: It's not obvious what `Ccn` means. Let's rename this class (and change the `NAME` value accordingly) to `CyclomaticComplexity`.

File: src/PhpMetricsChecker/Search/SearchesFactory.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?php
+declare(strict_types=1);
+
+namespace EonX\EasyQuality\PhpMetricsChecker\Search;
+
+use EonX\EasyQuality\PhpMetricsChecker\Metric\AbstractMetric;
+use Hal\Search\Search;
+use Hal\Search\Searches;
+
+final class SearchesFactory
```
Comment (roman-eonx) [inline src/PhpMetricsChecker/Search/SearchesFactory.php:10]: Does this `Factory` term come from the PhpMetrics library? Seeing the `build` method below, I'd suggest either renaming the class to `SearchesBuilder` or the method to `create`.

**Review by roman-eonx — CHANGES_REQUESTED** [review]

File: composer.json [hunk 3]
```diff
@@ -13,7 +13,8 @@
         "slevomat/coding-standard": "8.13.1",
         "symplify/easy-coding-standard": "11.5.0",
         "symplify/phpstan-extensions": "11.2.0",
-        "symplify/phpstan-rules": "12.0.4.72"
+        "symplify/phpstan-rules": "12.0.4.72",
+        "phpmetrics/phpmetrics": "^2.8"
```
[the remark this answers is not in the document]
Comment (ERuban) [reply composer.json]: done

File: src/PhpMetricsChecker/Metric/CcnMethodMax.php (hunk 1 above)
Comment (ERuban) [reply src/PhpMetricsChecker/Metric/CcnMethodMax.php]: done

File: src/PhpMetricsChecker/Search/SearchesFactory.php (hunk 2 above)
Comment (ERuban) [reply src/PhpMetricsChecker/Search/SearchesFactory.php:10]: Yes, Factory term is from `phpmetrics` library. And the method name in it is `factory()`. So have renamed the method name to `factory()` to me maximum consistent with phpmetrics

File: bin/pmc [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env php
+<?php
+error_reporting(error_reporting() & ~E_DEPRECATED);
```
Comment (itorgov) [inline bin/pmc:3]: Please add `declare(strict_types=1);`.

File: bin/pmc [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+use EonX\EasyQuality\PhpMetricsChecker\PhpMetricsChecker;
+
+function includeIfExists(string $file)
+{
+    if (\is_file($file)) {
+        return include $file;
```
Comment (itorgov) [inline bin/pmc:10]: Can we use the `require_once` operator instead?

File: src/PhpMetricsChecker/Config/ConfigFileReader.php [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+{
+    public function read(Config $config, string $fileName): void
+    {
+        $jsonText = \file_get_contents($fileName);
+
+        if ($jsonText === false) {
+            throw new InvalidArgumentException("Cannot read configuration file '$fileName'");
+        }
+
+        $jsonData = \json_decode($jsonText, true, 512, \JSON_THROW_ON_ERROR);
```
Comment (itorgov) [inline src/PhpMetricsChecker/Config/ConfigFileReader.php:20]: Let's use named arguments to skip the `depth` value:

```php
\json_decode(json: $jsonText, associative: true, flags: \JSON_THROW_ON_ERROR)
```

File: bin/pmc (hunk 4 above)
Comment (ERuban) [reply bin/pmc:4]: done

File: src/PhpMetricsChecker/Config/ConfigFileReader.php (hunk 6 above)
Comment (ERuban) [reply src/PhpMetricsChecker/Config/ConfigFileReader.php]: done

File: bin/pmc (hunk 5 above)
Comment (ERuban) [reply bin/pmc:11]: No, we cannot.
`[EA] Only the first call returns the proper/expected result. Subsequent calls will return 'true'.`

**Review by itorgov — APPROVED** [review]

**Review by roman-eonx — APPROVED** [review]

**Pull request merged**